src/nectarchain/dqm/pixel_participation.py

```diff
@@ -0,0 +1,148 @@
+            pixel = list(pixel)
+            pixels = np.concatenate([pixel21, pixel])
+        else: 
+            pixels = pixel
+
+        if evt.trigger.event_type.value == 32:  # count peds
+            self.counter_ped += 1
+            BadPixels_ped1 = list(map(int, pixelBAD[pixels]))
+            self.BadPixels_ped += BadPixels_ped1
+
+        else:
+            self.counter_evt += 1
+            BadPixels1 = list(map(int, pixelBAD[pixels]))
+            self.BadPixels += BadPixels1
+        return None
+
+    def FinishRun(self):
+        self.BadPixels_ped = np.array(self.BadPixels_ped)
+        self.BadPixels = np.array(self.BadPixels)
+
+
+
+    def GetResults(self):
+        # INITIATE DICT
+        self.PixelParticipation_Results_Dict = {}
+
+        # ASSIGN RESUTLS TO DICT
+        if self.k == 0:
+
+            if self.counter_evt > 0:
+                self.PixelParticipation_Results_Dict[
+                    "CAMERA-BadPix-PHY-OverEVENTS-HIGH-GAIN"
+                ] = self.BadPixels
+
+            if self.counter_ped > 0:
+                self.PixelParticipation_Results_Dict[
+                    "CAMERA-BadPix-PED-PHY-OverEVENTS-HIGH-GAIN"
+                ] = self.BadPixels_ped
+
+        if self.k == 1:
+            if self.counter_evt > 0:
+                self.PixelParticipation_Results_Dict[
+                    "CAMERA-BadPix-PHY-OverEVENTS-LOW-GAIN"
+                ] = self.BadPixels
+
+            if self.counter_ped > 0:
+                self.PixelParticipation_Results_Dict[
+                    "CAMERA-BadPix-PED-PHY-OverEVENTS-LOW-GAIN"
+                ] = self.BadPixels_ped
+
+        return self.PixelParticipation_Results_Dict
+
+    def PlotResults(self, name, FigPath):
+        self.PixelParticipation_Figures_Dict = {}
+        self.PixelParticipation_Figures_Names_Dict = {}
+
+        # titles = ['All', 'Pedestals']
+        if self.k == 0:
+            gain_c = "High"
+        if self.k == 1:
+            gain_c = "Low"
+
+        if self.counter_evt > 0:
+            fig1, self.disp1 = plt.subplots()
+            self.disp1 = CameraDisplay(
+                geometry=self.camera,
+                image=self.BadPixels,
+                cmap=self.cmap,
+            )
+            self.disp1.cmap = self.cmap
+            self.disp1.cmap = plt.cm.coolwarm
+            self.disp1.add_colorbar()
+            self.disp1.axes.text(2.0, 0, "Bad Pixels", rotation=90)
+            plt.title("Camera BPX %s gain (ALL)" % gain_c)
+
+            self.PixelParticipation_Figures_Dict[
+                "CAMERA-BADPIX-PHY-DISPLAY-%s-GAIN" % gain_c
+            ] = fig1
+            full_name = name + "_Camera_BPX_%sGain.png" % gain_c
+            FullPath = FigPath + full_name
+            self.PixelParticipation_Figures_Names_Dict[
+                "CAMERA-BADPIX-PHY-DISPLAY-%s-GAIN" % gain_c
+            ] = FullPath
+            plt.close()
+
+        if self.counter_ped > 0:
+            fig2, self.disp2 = plt.subplots()
```

Comment:
These lines are almost an exact duplication of lines 100+. Please try to regroup them together.

Reply:
this also applies for all the classes in dqm

Reply:
Done